decidim-core/app/controllers/decidim/participatory_processes_controller.rb

```diff
@@ -6,15 +6,14 @@ module Decidim
   # A controller that holds the logic to show ParticipatoryProcesses in a
   # public layout.
   class ParticipatoryProcessesController < ApplicationController
-    include NeedsParticipatoryProcess
-
     layout "layouts/decidim/participatory_process", only: [:show]
 
-    skip_after_action :verify_participatory_process, only: [:index]
+    before_action -> { extend(NeedsParticipatoryProcess) }, only: [:show]
```

Comment:
@josepjaume I'm not sure I understand your suggestion. Can you post a small pseudocode snippet with the idea or something like that?

Also, I should probably point out that I tried to split this change into a separate PR, but the other changes in here blow up with out it. This is because whereas before we used `find_by(id: my_id)` and then manually raise at specific points, we now use let `find(my_id)` let the exception be handled by Rails. That means that now, when calling `current_participatory_process` on `participatory_processes#index` we end up calling `find(nil)` and things blow up.

Finally, do you still use "review apps", seems like a good use case for one of those.